### PR TITLE
Update environment seeding code in tutorials

### LINKED ISSUE #1468: [Question] Inconsistency Between reset() Docstring and REINFORCE Tutorial: Seeding Every Episode
### Question

In the documentation for the `reset()` method of the `gym.Env` class, it states: "If you pass an integer, the PRNG will be reset even if it already exists. Usually, you want to pass an integer right after the environment has been initialized and then never again."

However, in the "Training using REINFORCE for Mujoco" tutorial, at line 263, it appears that the `np_random` Generator of the class is reinitialized at the beginning of every episode. This does not seem reasonable.

### PR DESCRIPTION
**Pull request by Trenza1ore**
# Description

1. Two tutorials in documentation have code that calls `env.reset(seed=seed)` for every reset, which is not the recommended practice as the environments will use the same random seed every episode, removing variety from training data.
2. Additionally, the reward logging of the `Training using REINFORCE for Mujoco` tutorial doesn't make sense (I confirmed with the current HEAD of main and after my change), as illustrated below, no reward is logged at episode 5000 (end of training), and episode 4952-5000's rewards bleed into episode 0's reward log of next run. This PR corrected this by applying the correct offset to the episode index and additionally printing a separator line to mark the current experiment's random seed.

**Current:**
```
Episode: 0 Average Reward: 8
Episode: 1000 Average Reward: 38
Episode: 2000 Average Reward: 83
Episode: 3000 Average Reward: 122
Episode: 4000 Average Reward: 150
Episode: 0 Average Reward: 235
Episode: 1000 Average Reward: 22
Episode: 2000 Average Reward: 59
Episode: 3000 Average Reward: 131
Episode: 4000 Average Reward: 159
Episode: 0 Average Reward: 216
Episode: 1000 Average Reward: 15
Episode: 2000 Average Reward: 36
Episode: 3000 Average Reward: 213
Episode: 4000 Average Reward: 865
Episode: 0 Average Reward: 856
Episode: 1000 Average Reward: 19
Episode: 2000 Average Reward: 51
Episode: 3000 Average Reward: 138
Episode: 4000 Average Reward: 382
Episode: 0 Average Reward: 908
Episode: 1000 Average Reward: 20
Episode: 2000 Average Reward: 63
Episode: 3000 Average Reward: 145
Episode: 4000 Average Reward: 229
```
**Fixed:**
```
--------- Random Seed 1 ---------
Episode: 1000 Average Reward: 40
Episode: 2000 Average Reward: 97
Episode: 3000 Average Reward: 125
Episode: 4000 Average Reward: 177
Episode: 5000 Average Reward: 273
--------- Random Seed 2 ---------
Episode: 1000 Average Reward: 20
Episode: 2000 Average Reward: 73
Episode: 3000 Average Reward: 132
Episode: 4000 Average Reward: 177
Episode: 5000 Average Reward: 244
--------- Random Seed 3 ---------
Episode: 1000 Average Reward: 17
Episode: 2000 Average Reward: 42
Episode: 3000 Average Reward: 235
Episode: 4000 Average Reward: 862
Episode: 5000 Average Reward: 832
--------- Random Seed 5 ---------
Episode: 1000 Average Reward: 19
Episode: 2000 Average Reward: 51
Episode: 3000 Average Reward: 122
Episode: 4000 Average Reward: 353
Episode: 5000 Average Reward: 882
--------- Random Seed 8 ---------
Episode: 1000 Average Reward: 16
Episode: 2000 Average Reward: 55
Episode: 3000 Average Reward: 129
Episode: 4000 Average Reward: 263
Episode: 5000 Average Reward: 787
```

Fixes #1468 

## Type of change

Please delete options that are not relevant.

- [x] Documentation only change (no code changed)
- [x] Bug fix (non-breaking change which fixes an issue)

#### Solving Frozenlake with Tabular Q-Learning
Figure_1
Figure_2

#### Training using REINFORCE for Mujoco (New vs Old)
Figure_3
Figure_4

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/tutorials/training_agents/frozenlake_q_learning.py [hunk 1]
```diff
@@ -185,11 +185,11 @@ def run_env():
 
     for run in range(params.n_runs):  # Run several times to account for stochasticity
         learner.reset_qtable()  # Reset the Q-table between runs
+        state = env.reset(seed=params.seed)[0]  # Reset the environment with set seed
```
Comment (pseudo-rnd-thoughts) [inline docs/tutorials/training_agents/frozenlake_q_learning.py]: state, info = env.reset(seed=params.seed)

File: docs/tutorials/training_agents/frozenlake_q_learning.py [hunk 2]
```diff
@@ -221,6 +221,9 @@ def run_env():
             # Log all rewards and steps
             rewards[episode, run] = total_rewards
             steps[episode, run] = step
+
+            # Reset the environment
+            state = env.reset()[0]
```
Comment (pseudo-rnd-thoughts) [inline docs/tutorials/training_agents/frozenlake_q_learning.py]: state, info = env.reset()

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Thanks for the PR, otherwise, looks good

File: docs/tutorials/training_agents/frozenlake_q_learning.py (hunk 1 above)
Comment (Trenza1ore) [reply docs/tutorials/training_agents/frozenlake_q_learning.py]: Made the change :-)

File: docs/tutorials/training_agents/frozenlake_q_learning.py (hunk 2 above)
Comment (Trenza1ore) [reply docs/tutorials/training_agents/frozenlake_q_learning.py]: Made the change here as well

**Comment by Trenza1ore** [comment]
> Thanks for the PR, otherwise, looks good

Made the requested changes 🙂

**Review by pseudo-rnd-thoughts — APPROVED** [review]
LGTM

**Review by pseudo-rnd-thoughts — COMMENTED** [review]
@Trenza1ore Should we be updating the tutorial images if this has a performance change?

**Comment by Trenza1ore** [comment]
> @Trenza1ore Should we be updating the tutorial images if this has a performance change?

That would be a good idea for frozen lake and mujoco both. But also I think the mujoco change is mostly PyTorch version differences  (as you can see in PR description, the new and old version of mujoco REINFORCE code with current PyTorch yields almost the same results but the original plot from tutorial is very different). 

Maybe there should be a second PR updating result images for all Training Agents tutorials? @pseudo-rnd-thoughts 

Original plot:
image

**Comment by pseudo-rnd-thoughts** [comment]
Nahh, just update them here @Trenza1ore

**Comment by Trenza1ore** [comment]
> Nahh, just update them here @Trenza1ore 

Will do that later today then 👍

**Comment by Trenza1ore** [comment]
@pseudo-rnd-thoughts I was unable to capture the exact UI & dimensions of the images, as I believe that those are {OS / screen resolution / matplotlib backend}-dependent. I tried my best to resize, align & crop, and the result is different albeit "not too bad" (New vs Original):
Screenshot 2026-07-01 at 9 38 42 pm
Screenshot 2026-07-01 at 9 47 02 pm

**Review by pseudo-rnd-thoughts — APPROVED** [review]
LGTM
